internal/k8s/k8s.go

```diff
@@ -435,3 +438,20 @@ func (c *Client) sync(key interface{}) error {
 		panic(fmt.Errorf("unknown key type for %#v (%T)", key, key))
 	}
 }
+
+// InClusterNamespace retrieves the namespace metallb is running in, when
+// running in-cluster.
+func InClusterNamespace() (string, error) {
```

Comment:
This is based on [kubernetes/client-go](https://github.com/kubernetes/client-go/blob/master/tools/clientcmd/client_config.go#L497). I have briefly investigated if we should build the k8s client in a way that gives access to this function but did not want to alter too much code in `internal/k8s/`. I believe that this is currently the better solution.